File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string solution = "";
        char ltr = strs[0][0];

        if(strs.size() <= 0){
            return "";
        }

        for(int i = 0; i < strs[0].size(); i++){
            ltr = strs[0][i];
            for(int j = 0; j < strs.size(); j++){
                if(strs[j][i] != ltr){
                    return solution;
                }
            }
            solution += ltr;
        }
        return solution;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        newDict = defaultdict(list)
        rand = []
        for s in strs:
            sortedS = ''.join(sorted(s))
            newDict[sortedS].append(s)
        
        return list(newDict.values())